libfreefare/freefare.c

```diff
@@ -103,10 +103,21 @@ freefare_get_tags(nfc_device *device)
 	}
     }
 
-    // Poll for a FELICA tag
+    // Poll for a FELICA 424 kbit tag
     modulation.nmt = NMT_FELICA;
-    modulation.nbr = NBR_424; // FIXME NBR_212 should also be supported
-    if ((candidates_count = nfc_initiator_list_passive_targets(device, modulation, candidates, MAX_CANDIDATES)) < 0)
+    modulation.nbr = NBR_424;
+
+    candidates_count = nfc_initiator_list_passive_targets(device, modulation, candidates, MAX_CANDIDATES);
+    if (candidates_count < 0)
+	return NULL;
+
+    // Poll for a FELICA 212 kbit tag (only if no 424 kbit tag was found)
```

Comment:
Can't we list both 212 and 424 kbit tags instead of listing 212 tags when no 424 is found?

Reply:
I guess it does: as far as I can recall, I ran the test suite with 2 devices and 1 or 2 cards on each.  Has less communication problems than with 1 device and 3+ tags :smile: 

But sometimes my memory tricks on me, and I don't have this hardware anymore to check.